pkg/rke2/psp.go

```diff
@@ -0,0 +1,449 @@
+				logrus.Infof("Setting Cluster RoleBinding: %s", systemUnrestrictedRoleBindingName)
+				tmpl := fmt.Sprintf(systemUnrestrictedNodesRoleBindingTemplate, systemUnrestrictedRoleBindingName, systemUnrestrictedRoleName)
+				if err := deployClusterRoleBindingFromYaml(ctx, cs, tmpl); err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+		if _, err := cs.RbacV1().ClusterRoleBindings().Get(ctx, systemUnrestrictedSvcAcctRoleBindingName, metav1.GetOptions{}); err != nil {
+			if apierrors.IsNotFound(err) {
+				logrus.Infof("Setting Cluster RoleBinding: %s", systemUnrestrictedSvcAcctRoleBindingName)
+				tmpl := fmt.Sprintf(systemUnrestrictedServiceAcctRoleBindingTemplate, systemUnrestrictedSvcAcctRoleBindingName, systemUnrestrictedRoleName)
+				if err := deployRoleBindingFromYaml(ctx, cs, tmpl); err != nil {
+					return err
+				}
+			} else {
+				return err
+			}
+		}
+		annotations[namespaceAnnotationSystemUnrestricted] = cisAnnotationValue
+	}
+	return nil
+}
+
+// setPSPs sets the default PSP's based on the mode that RKE2 is running in. There is either CIS or non
+// CIS mode. For CIS mode, the globalRestricted and systemUnrestricted polices and associated roles and
+// bindings are created. For non CIS mode, globalRestricted, system-unrestricted, and global-unrestricted
+// policies and associated roles and bindings are loaded, if applicable.
+//
+// Each PSP has an annotation associated with it that we check to see exists before performing any
+// write operations. The annotation is created upon successful setting of PSPs.
+//
+// Load logic
+//
+// CIS:
+// - If the globalRestricted annotation does not exist, create PSP, role, binding.
+// - If the systemUnrestricted annotation does not exist, create the PSP, role, binding.
+// - If the globalUnrestricted annotation does not exist, check if PSP exists, and if so,
+//   delete it, the role, and the bindings.
+//
+// non-CIS:
+// - If the globalUnrestricted annotation does not exist, then create PSP, role, binding.
+// - If the systemUnrestricted annotation does not exist, then create PSP, role, binding.
+// - If the globalRestricted annotation does not exist, then check if the PSP exists and
+//   if it doesn't, create it. Check if the associated role and bindings exist and
+//   if they do, delete them.
+func setPSPs(clx *cli.Context) func(context.Context, daemonsConfig.Control) error {
+	return func(ctx context.Context, cfg daemonsConfig.Control) error {
+		logrus.Info("Applying PSP's...")
+		go func() {
+			<-cfg.Runtime.APIServerReady
+
+			cs, err := newClient(cfg.Runtime.KubeConfigAdmin, nil)
+			if err != nil {
+				logrus.Fatalf("psp: new k8s client: %s", err.Error())
+			}
+
+			ns, err := cs.CoreV1().Namespaces().Get(ctx, metav1.NamespaceSystem, metav1.GetOptions{})
+			if err != nil {
+				logrus.Fatalf("psp: get kube-system namespace: %s", err.Error())
+			}
+
+			annotations := make(map[string]string)
+
+			logrus.Info("Applying PSP's...")
```

Comment:
you have the same log message twice (here and line 158). Any way we can disambiguate (or delete one)?